### PR TITLE
CI | Data Buckets | Test AWS SDK Clients

### PR DESCRIPTION
**Pull request by shirady**
### Explain the changes
1. Add to the CI a test that runs basic commands using the most updated AWS SDK client. In this PR it is AWS SDK Go V2 as a start.
2. Add log printings of `'pools_to_create.length'` in coretest (helps in debugging in case the nodes/hosts were not created).

### Issues:
1. The go script and its test were described in PR noobaa/noobaa-operator#1521 and the need for having a CI that runs with most updated SDK version was raised as a part of PR #8753 to help us know of major changes in the AWS client behavior.

GAP - we need to add a step to report the nightly test in our Slack channel (currently the secret is empty, planning to do it in another PR) #8905  

### Testing Instructions:
1. Run: `make test-aws-sdk-clients`

- [ ] Doc added/updated
- [X] Tests added

### CODE REVIEW

File: src/test/unit_tests/different_clients/test_go_sdkv2_script.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/* Copyright (C) 2025 NooBaa */
+'use strict';
+
+// disabling init_rand_seed as it takes longer than the actual test execution
+process.env.DISABLE_INIT_RANDOM_SEED = "true";
+
+// setup coretest first to prepare the env
+const coretest = require('../coretest');
+coretest.setup({ pools_to_create: [coretest.POOL_LIST[1]] });
```
Comment (achouhan09) [inline src/test/unit_tests/different_clients/test_go_sdkv2_script.js:9]: Can you explain in short, why `pool-with-10-hosts` and its usage?

File: src/test/unit_tests/different_clients/test_go_sdkv2_script.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            const output = await run_go_sdk_v2_client_script(bucket_name, key_name, mpu_key_name, endpoint);
+            console.log('\nTest output:\n');
+            console.log(output);
+            // here we will fail the test in case of at least one failing test printing
+            const match_failing_tests_statement = output.match(/Failing Tests:\s*(\d+)/i);
+            // we don't see the failing tests output we will set a number higher than 1 to fail the test on purpose
+            const number_of_failing_tests = match_failing_tests_statement ? parseInt(match_failing_tests_statement[1], 10) : 1;
+            assert.equal(number_of_failing_tests, 0);
+        } catch (err) {
+            console.log('while execution of aws go client had the err', err);
```
Comment (achouhan09) [inline src/test/unit_tests/different_clients/test_go_sdkv2_script.js:48]:
```suggestion
            console.log('error during the aws go client execution: ', err);
```

File: src/test/unit_tests/different_clients/go_aws_sdkv2_client.go [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+		skippedTests = append(skippedTests, "completeMultiPartUpload")
+	}
+}
+
+// deleteObjectsAndBucket will delete the 2 created objects (one from Put Objects and the other from Multipart Upload Objects) and the bucket
+func deleteObjectsAndBucket(client *s3.Client) {
+	if disableDeletion {
+		fmt.Println("--------------------------------------------------")
+		fmt.Println("disable-deletion flag was set, will not operate delete commands")
+	}
```
Comment (achouhan09) [inline src/test/unit_tests/different_clients/go_aws_sdkv2_client.go:147]: I think we can return here, if deletion is disabled to avoid later checks.

File: src/test/unit_tests/different_clients/test_go_sdkv2_script.js (hunk 1 above)
Comment (shirady) [reply src/test/unit_tests/different_clients/test_go_sdkv2_script.js:9]: In this test, I'm using `name: 'pool-with-1-host'` (and not `name: 'pool-with-10-hosts'`), as I need a minimal check and I can use only 1 node. Probably tests that are focused on hosts will use more hosts for testing, in this case, it is basic.

File: src/test/unit_tests/different_clients/test_go_sdkv2_script.js (hunk 2 above)
Comment (shirady) [reply src/test/unit_tests/different_clients/test_go_sdkv2_script.js]: Changed, thanks.

File: src/test/unit_tests/different_clients/go_aws_sdkv2_client.go [hunk 4]
```diff
@@ -0,0 +1,345 @@
...
+		completedParts := uploadPart(client, bucket, mpu, uploadId)
+		if completedParts != nil {
+			completeMultiPartUpload(client, bucket, mpu, uploadId, completedParts)
+		} else {
+			skippedTests = append(skippedTests, "completeMultiPartUpload")
+		}
+	} else {
+		skippedTests = append(skippedTests, "uploadPart")
+		skippedTests = append(skippedTests, "completeMultiPartUpload")
+	}
```
[the remark this answers is not in the document]
Comment (shirady) [reply src/test/unit_tests/different_clients/go_aws_sdkv2_client.go:144]: Yes, I moved it to a function, thanks.

File: src/test/unit_tests/different_clients/go_aws_sdkv2_client.go (hunk 3 above)
Comment (shirady) [reply src/test/unit_tests/different_clients/go_aws_sdkv2_client.go]: I added an else clause, thanks.

**Review by aayushchouhan09 — APPROVED** [review]
LGTM, Please remember to update the CI timeline to nightly instead of push/pull if we are plainning it in the same PR.
